src/agent/reqwest-retry/src/lib.rs

```diff
@@ -172,6 +172,19 @@ impl SendRetry for reqwest::RequestBuilder {
     }
 }
 
+pub fn is_auth_failure(response: &Result<Response>) -> bool {
+    if let Err(error) = response {
+        if let Some(error) = error.downcast_ref::<reqwest::Error>() {
+            if let Some(status) = error.status() {
+                if status == StatusCode::UNAUTHORIZED {
+                    return true;
+                }
+            }
+        }
```

Comment:
his only check the case where the request was converted to an error with a call to `error_for_status` most likely.
We should also check the response status code without the downcast too.

```suggestion
        if let Some(error) = error.downcast_ref::<reqwest::Error>() {
            if let Some(status) = error.status() {
                if status == StatusCode::UNAUTHORIZED {
                    return true;
                }
            }
        } else {
            return response.status() == StatusCode::UNAUTHORIZED;
        }
```

Reply:
For calls from reqwest-retry, always done within the context of error_for_status.

However, I see the value of adding it.  Note, it needs to be done up a level.  I've pushed an update to add this functionality.